includes/class-wc-subscriptions-data-copier.php

```diff
@@ -0,0 +1,526 @@
+		'_order_key',
+		'_edit_lock',
+		'_wc_points_earned',
+		'_transaction_id',
+		'_billing_interval',
+		'_billing_period',
+		'_subscription_resubscribe',
+		'_subscription_renewal',
+		'_subscription_switch',
+		'_payment_method',
+		'_payment_method_title',
+		'_suspension_count',
+		'_requires_manual_renewal',
+		'_cancelled_email_sent',
+		'_trial_period',
+		'_created_via',
+		'_order_stock_reduced',
+	];
+
+	/**
+	 * The subscription or order being copied.
+	 *
+	 * @var WC_Order
+	 */
+	private $from_object = null;
+
+	/**
+	 * The subscription or order being copied to.
+	 *
+	 * @var WC_Order
+	 */
+	private $to_object = null;
+
+	/**
+	 * The type of copy. Can be 'subscription' or 'renewal'.
+	 *
+	 * Used in dynamic filters to allow third parties to target specific meta keys in different copying contexts.
+	 *
+	 * @var string
+	 */
+	private $copy_type = '';
+
+	/**
+	 * Copies data from one object to another.
+	 *
+	 * This function acts as a publicly accessible wrapper for obtaining an instance of the copier and completing the copy.
+	 *
+	 * @param WC_Order $from_object The object to copy data from.
+	 * @param WC_Order $to_object   The object to copy data to.
+	 * @param string   $copy_type   Optional. The type of copy. Can be 'subscription', 'parent', 'renewal_order' or 'resubscribe_order'. Default is 'subscription'.
+	 */
+	public static function copy( $from_object, $to_object, $copy_type = self::DEFAULT_COPY_TYPE ) {
+		$instance = new self( $from_object, $to_object, $copy_type );
+		$instance->copy_data();
+	}
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Order $from_object The object to copy data from.
+	 * @param WC_Order $to_object   The object to copy data to.
+	 * @param string   $copy_type   Optional. The type of copy. Can be 'subscription', 'parent', 'renewal_order' or 'resubscribe_order'. Default is 'subscription'.
+	 */
+	public function __construct( $from_object, $to_object, $copy_type = self::DEFAULT_COPY_TYPE ) {
+		$this->from_object = $from_object;
+		$this->to_object   = $to_object;
+		$this->copy_type   = $copy_type;
+	}
+
+	/**
+	 * Copies the data from the "from" object to the "to" object.
+	 */
+	public function copy_data() {
+
+		if ( $this->is_using_cpt_data_stores() ) {
+			$data_array = $GLOBALS['wpdb']->get_results( $this->get_deprecated_meta_query(), ARRAY_A );
+			$data       = wp_list_pluck( $data_array, 'meta_value', 'meta_key' );
+		} else {
+			$data  = $this->get_meta_data();
+			$data += $this->get_order_data();
+			$data += $this->get_operational_data();
+			$data += $this->get_address_data();
+
+			// Remove any excluded meta keys.
+			$data = $this->filter_excluded_meta_keys_via_query( $data );
+		}
+
+		$data = $this->apply_deprecated_filter( $data );
+
+		/**
+		 * Filters the data to be copied from one object to another.
+		 *
+		 * This filter name contains a dynamic part, $this->copy_type, which can be one of:
+		 * 'subscription', 'parent', 'renewal_order' or 'resubscribe_order'.
```

Comment:
Recommend spelling these out explicitly so they are searchable, e.g. 

`wc_subscriptions_subscription_data` `wc_subscriptions_parent_data` `wc_subscriptions_renewal_order_data` `wc_subscriptions_resubscribe_order_data`

Reply:
Searchability is something I can definitely get behind. I've exploded the lists in [b203673](https://github.com/Automattic/woocommerce-subscriptions-core/pull/229/commits/b2036730d5fcc8745e2c0a164d43c8b4a9be0ddb).